app/src/main/java/de/westnordost/streetcomplete/overlays/street_furniture/StreetFurnitureOverlayForm.kt

```diff
@@ -0,0 +1,159 @@
+import de.westnordost.streetcomplete.util.ktx.viewLifecycleScope
+import de.westnordost.streetcomplete.view.controller.FeatureViewController
+import de.westnordost.streetcomplete.view.dialogs.SearchFeaturesDialog
+import kotlinx.coroutines.launch
+
+class StreetFurnitureOverlayForm : AbstractOverlayForm() {
+
+    override val contentLayoutResId = R.layout.fragment_overlay_street_furniture
+    private val binding by contentViewBinding(FragmentOverlayStreetFurnitureBinding::bind)
+
+    private var originalFeature: Feature? = null
+
+    private lateinit var featureCtrl: FeatureViewController
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+
+        val element = element
+        originalFeature = element?.let {
+            if (IS_DISUSED_STREET_FURNITURE_EXPRESSION.matches(element)) {
+                DummyFeature(
+                    "street_furniture/unknown_disused",
+                    requireContext().getString(R.string.unknown_disused_street_furniture),
+                    "ic_preset_maki_marker_stroked",
+                    element.tags
+                )
+            } else {
+                featureDictionary
+                    .byTags(element.tags)
+                    .forLocale(*getLocalesForFeatureDictionary(resources.configuration))
+                    .forGeometry(element.geometryType)
+                    .inCountry(countryOrSubdivisionCode)
+                    .find()
+                    .firstOrNull()
+                // if not found anything in the iD presets, then something weird happened
+                ?: DummyFeature(
+                    "street_furniture/unknown",
+                    requireContext().getString(R.string.unknown_street_furniture),
+                    "ic_preset_maki_marker_stroked",
+                    element.tags
+                )
+            }
+        }
+    }
+
+    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
+        super.onViewCreated(view, savedInstanceState)
+
+        // title hint label with name is a duplication, it is displayed in the UI already
+        setTitleHintLabel(element?.tags?.let { getLocationLabel(it, resources) })
+        setMarkerIcon(R.drawable.ic_quest_apple)
+
+        featureCtrl = FeatureViewController(featureDictionary, binding.featureTextView, binding.featureIconView)
+        featureCtrl.countryOrSubdivisionCode = countryOrSubdivisionCode
+        featureCtrl.feature = originalFeature
+
+        binding.featureView.setOnClickListener {
+            SearchFeaturesDialog(
+                requireContext(),
+                featureDictionary,
+                element?.geometryType,
+                countryOrSubdivisionCode,
+                featureCtrl.feature?.name,
+                ::filterOnlyStreetFurniture,
+                ::onSelectedFeature,
+                listOf(
+                    // ordered by popularity, skipping trees as there are multiple variants of them
```

Comment:
should be ordered by occurrence (not necessarily the current popularity). Hence, makes sense that street lamp is at the top, but so should `natural=tree`. What variants?

(Whether or not both trees and street lamps should be moved to their own layer - in the latter case, the street lighting layer? - is independent of that.)

(Also, this should probably rather be a private val at top of the class)

Reply:
Broadleaf, needlelaved and so on seems to have own entries in addition to general tree entry.

If someone is mapping trees at all then specific variants are better than general ones

Reply:
Why does it need to have own categories? Could one not simply have `natural/tree` here? I think this should be possible with the iD presets. Also, I guess if one **had** to choose, it would be the broadleaf one, as these are much more common as street trees.

Reply:
`natural/tree` can go here but that works encourage mappers to use that instead of more precise ones (they may be even unaware of them and be stuck with general entry now as latest used one)

Adding broadleaf deciduous one only can cause them to use it also for other cases.

Adding all variants would take too much space.

Reply:
Some data:

- About 4 million street lanterns are mapped.
- About 23 million trees are mapped.
- Only 18% (4 million) trees are tagged with `leaf_type=*`. However, of these, 92% (4 million) are tagged with `broadleaved`
- iD presets exist for all of *tree*, *broadleaved tree* and *needleleaved tree*, each named properly. The icon for the latter is a needleaved tree, the icons for the former are (two different!) broadleaved trees

Reply:
I made a list of further things that could be in the quick select.

```
tree				23.0M
broadleaved tree		 4.0M
needleleaved tree		 0.3M

waste basket			 0.7M (fast growth)

recycling container		 0.4M

artwork				 0.2M

charging station		 0.1M (fast growth)
```

It is most convenient, if things that would often be added by users was in this list.

For this reason I suggest to remove the defibrillator (only at commercial buildings) and picnic table (only in parks) and add both the broadleaved tree and the needleleaved tree. Not sure about the others and how much space there is (on small screen)